File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Code/Permissao/PermissaoDAL.cs
using JsonWebToken.Model;
using System;
using System.Collections.Generic;
using System.Data.Common;
using System.Data;
using Oracle.ManagedDataAccess.Client;
using JsonWebToken.Util;

namespace JsonWebToken.Code.Permissao
{
  public class PermissaoDAL
  {
    public List<PermissaoModel> SelectPermissao(List<string> lstNomGrpAcs)
    {
      var permissoes = new List<PermissaoModel>();
      var lstUpdNomGrpAcs = new List<string>();

      foreach (var item in lstNomGrpAcs)
      {
        lstUpdNomGrpAcs.Add(string.Format("'{0}'", item));
      }

      var strNomGrpAcs = string.Join(",", lstUpdNomGrpAcs);

      if (lstNomGrpAcs.Count == 0 || string.IsNullOrEmpty(strNomGrpAcs))
        return permissoes;

      DbCommand databaseCommand = null;
      OracleCommand command = null;

      try
      {
        PermissaoSQL sqlCmd = new PermissaoSQL();

        command = new OracleCommand(sqlCmd.SelectPermissao(strNomGrpAcs), new OracleConnection(Utils.ORACLE_CONNECCAO));

        command.Connection.Open();

        DataTable dt = new DataTable();
        dt.Load(command.ExecuteReader());

        foreach (DataRow row in dt.Rows)
        {
          permissoes.Add(new PermissaoModel
          {
            CODACSFUN = row.Field<long>("CODACSFUN"),
            NOMACSFUN = row.Field<String>("NOMACSFUN")
          });
        }
      }
      catch (Exception ex)
      {
        throw ex;
      }
      finally
      {
        if (command != null && command.Connection.State == ConnectionState.Open)
          command.Connection.Close();

        if (databaseCommand != null && databaseCommand.Connection != null && databaseCommand.Connection.State == ConnectionState.Open)
          databaseCommand.Connection.Close();
      }

      return permissoes;
    }
  }
}

File: Backend/Martins.Conciliacao.API/Code/Permissao/PermissaoDAL.vb
Imports Oracle.ManagedDataAccess.Client
Imports Martins.Conciliacao.Model
Imports Martins.Conciliacao.Util

Namespace Code.Permissao
	Public Class PermissaoDAL
		Public Function SelectPermissao(lstNomGrpAcs As List(Of String)) As List(Of PermissaoModel)
			Dim permissoes = New List(Of PermissaoModel)()
			Dim lstUpdNomGrpAcs = New List(Of String)()

            For Each item As Object In lstNomGrpAcs
                lstUpdNomGrpAcs.Add(String.Format("'{0}'", item))
            Next

            Dim strNomGrpAcs = String.Join(",", lstUpdNomGrpAcs)

			If lstNomGrpAcs.Count = 0 OrElse String.IsNullOrEmpty(strNomGrpAcs) Then
				Return permissoes
			End If

      Dim databaseCommand As OracleCommand = Nothing
      Dim command As OracleCommand = Nothing

			Try
				Dim sqlCmd As New PermissaoSQL()

				command = New OracleCommand(sqlCmd.SelectPermissao(strNomGrpAcs), New OracleConnection(Utils.ORACLE_CONNECCAO))

				command.Connection.Open()

				Dim dt As New DataTable()
				dt.Load(command.ExecuteReader())

				For Each row As DataRow In dt.Rows
          permissoes.Add(New PermissaoModel() With {
                        .CODACSFUN = row.Field(Of Long)("CODACSFUN"),
                        .NOMACSFUN = row.Field(Of [String])("NOMACSFUN")
                    })
        Next
			Catch ex As Exception
				Throw ex
			Finally
				If command IsNot Nothing AndAlso command.Connection.State = ConnectionState.Open Then
					command.Connection.Close()
				End If

				If databaseCommand IsNot Nothing AndAlso databaseCommand.Connection IsNot Nothing AndAlso databaseCommand.Connection.State = ConnectionState.Open Then
					databaseCommand.Connection.Close()
				End If
			End Try

			Return permissoes
		End Function
	End Class
End Namespace
